lib/files/index.js

```diff
@@ -0,0 +1,120 @@
+const { mkdir, readdir, rm, rename, stat, writeFile } = require('fs/promises')
+const { existsSync }  = require('fs')
+const { dirname, join, normalize } = require('path')
+const multer = require('multer')
+
+const  filesInterface = (app, settings)  => {
+    const BASE_PATH = join(settings.rootDir, settings.userDir, 'storage')
+    // not sure about this yet, need to check how file storage works
+    const storage = multer.memoryStorage()
+    const upload = multer({storage: storage})
+
+    const regex = /^\/flowforge\/files\/_\/([\.a-zA-Z0-9\/\-_ ]*)$/
+
+    const listDirectory = async (path = '') => {
+        const fullPath = normalize(join(BASE_PATH, path))
+        if (fullPath.startsWith(BASE_PATH)) {
+            const files = await readdir(fullPath, {withFileTypes: true})
+            const response = {
+                meta: {
+                    next_cursor: ''
+                },
+                files: [],
+                count: files.length
+            }
+            for (i in files) {
+                const file = files[i]
+                const fullPath = join(file.path, file.name)
+                const s = await stat(fullPath)
+                const rep = {
+                    name: file.name,
+                    lastModified: s.mtime
+                }
+                if (file.isFile()) {
+                    rep.type = 'file'
+                    rep.size = s.size
+                    
+                } else if (file.isDirectory()) {
+                    rep.type = 'directory'
+                }
+                response.files.push(rep)
+            }
+            return response
+        } else {
+            return []
+        }
+    }
+
+    app.get(regex, async (request, reply) => {
+        try {
+            const files = await listDirectory(request.params[0])
+            reply.send(files)
+        } catch (err) {
+            if (err.code === 'ENOENT') {
+                reply.status(404).send()
+            }
+        }
```

Comment:
This works fine for listing a directory. If I try accessing an individual file, I get no response at all - the request hangs. I assume an error is being thrown, but not ENOENT, so the reply never gets set.

For now, lets 404 it - but that may need to change before release.

This is an open question on the api in terms of retrieving a file. I'm hesitant to just say a get request should return the file contents - its the obvious choice, but it feels too easy to get wrong somehow. Cannot quite put my finger on it.

Reply:
It replies with `ENOTDIR`, but the spec said it explicitly wouldn't be used to get a file, so it doesn't try and return them. I'll try and find a suitable 4xx code for wrong type.